File: aoj/kmc2022_n_9/a/cpp/main.cpp
#include <iostream>
#include <vector>

using std::cin;
using std::cout;
using std::endl;

int main() {
  while (true) {
    int32_t in_n(0);
    int32_t in_m(0);
    cin >> in_n >> in_m;
    if (in_n == 0 && in_m == 0) {
      break;
    }
    auto a_vec = std::vector<int32_t>(in_n);
    for (auto &a : a_vec) {
      cin >> a;
    }

    int32_t best_sum(0);

    for (int32_t i = 0; i < in_n - 1; ++i) {
      for (int32_t j = i + 1; j < in_n; ++j) {
        int32_t sum = a_vec[i] + a_vec[j];
        if (best_sum < sum && sum <= in_m) {
          best_sum = sum;
        }
      }
    }

    if (best_sum == 0) {
      cout << "NONE" << endl;
    } else {
      cout << best_sum << endl;
    }
  }
  return EXIT_SUCCESS;
}

File: aoj/kmc2022_n_9/a/rust/src/main.rs
fn main() {
    loop {
        // proconio::input! {
        input! {
            in_n: usize,
            in_m: usize,
        }
        if in_n == 0 && in_m == 0 {
            break;
        }
        // proconio::input! {
        input! {
            in_a_vec: [i32; in_n],
        }


        let mut best_sum: i32 = 0;

        for i in 0..in_n {
            for j in i+1..in_n {
                let sum = in_a_vec[i] + in_a_vec[j];
                if best_sum < sum && sum <= in_m as i32 {
                    best_sum = sum;
                }
            }
        }

        if best_sum == 0 {
            println!("NONE");
        } else {
            println!("{}", best_sum);
        }
    }
}
